fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java

```diff
@@ -114,20 +116,105 @@ public class TransactionLoadAction extends RestBaseAction {
     // Map operation name to metrics
     private final Map<TransactionOperation, OpMetrics> opMetricsMap = new HashMap<>();
 
-    private final ReadWriteLock txnNodeMapAccessLock = new ReentrantReadWriteLock();
-    private final Map<String, Long> txnNodeMap = new LinkedHashMap<>(512, 0.75f, true) {
-        @Override
-        protected boolean removeEldestEntry(Map.Entry<String, Long> eldest) {
-            return size() > (GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalBackendNumber() +
-                    GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalComputeNodeNumber()) * 512;
-        }
-    };
+    private final TransactionLoadLabelCache txnNodeMap = new TransactionLoadLabelCache();
 
     public TransactionLoadAction(ActionController controller) {
         super(controller);
         initMetrics();
     }
 
+    public static class TransactionLoadLabelCache  {
+        private final Cache<String, Long> cache;
+
+        public TransactionLoadLabelCache() {
+            Caffeine<Object, Object> builder = Caffeine.newBuilder();
+            cache = builder.initialCapacity(512).maximumWeight(getMaxCapacity())
+                    .weigher((key, value) -> 1)
+                    .expireAfterAccess((long) Config.prepared_transaction_default_timeout_second + 100, TimeUnit.SECONDS)
```

Comment:
`Config.prepared_transaction_default_timeout_second` is a mutable configuration, once the configuration is changed to a much longer time in runtime than the one `TransactionLoadLabelCache` is initialized here, it will cause the cache expired unexpectedly due to the fixed expiration here.